src/perception/long_range_cam/long_range_cam.cpp

```diff
@@ -0,0 +1,86 @@
+#include "long_range_cam.hpp"
+#include <opencv2/imgproc.hpp>
+#include <opencv2/videoio.hpp>
+#include <ros/init.h>
+#include <ros/this_node.h>
+#include <ros/time.h>
+#include <sensor_msgs/image_encodings.h>
+#include <thread>
+
+namespace mrover {
+
+    using namespace std::chrono_literals;
+
+    /**
+    * @brief Load config, open the camera, and start our threads
+    */
+    void LongRangeCamNodelet::onInit() {
+        try {
+            mGrabThread = std::jthread(&LongRangeCamNodelet::grabUpdate, this);
+        } catch (std::exception const& e) {
+            NODELET_FATAL("Exception while starting: %s", e.what());
+            ros::shutdown();
+        }
+    }
+
+    void fillImageMessage(cv::Mat& bgra, sensor_msgs::ImagePtr const& msg) {
+        assert(msg);
+
+        msg->height = bgra.rows;
+        msg->width = bgra.cols;
+        msg->encoding = sensor_msgs::image_encodings::BGR8;
+        msg->step = bgra.step[0];
+        msg->is_bigendian = __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__;
+        auto* imgPtr = bgra.ptr<float>(0);
+        size_t size = msg->step * msg->height;
+        msg->data.resize(size);
+        memcpy(msg->data.data(), imgPtr, size);
+    }
+
+    void LongRangeCamNodelet::grabUpdate() {
+        // See: http://wiki.ros.org/roscpp/Overview/NodeHandles for public vs. private node handle
+        // MT means multithreaded
+        mNh = getMTNodeHandle();
+        mPnh = getMTPrivateNodeHandle();
+        mCamInfoPub = mNh.advertise<sensor_msgs::CameraInfo>("long_range_cam/camera_info", 1);
+        mImgPub = mNh.advertise<sensor_msgs::Image>("long_range_image", 1);
+        // While dtor not called
+        cv::VideoCapture mCapture{std::format("v4l2src device=/dev/arducam ! videoconvert ! video/x-raw,width={},height={},format=I420,framerate={}/1 ! appsink", 1920, 1080, 5), cv::CAP_GSTREAMER};
```

Comment:
I would make the device file configurable, same with the width and height (use rosparams). That way we can use this node for the other USB cams.